### PR TITLE
fix channels priorities

### LINKED ISSUE #241: BUG: issue with pacakges installed from conda-forge in the Lasso bench
this bug happens in the lasso benchmark but I believe it's a benchopt issue. 

It seems conda does not use conda-forge as a channel inside the env (thus it fails to find glum and others):

```
(benchopt_benchmark_lasso) ➜  benchmark_lasso git:(main) conda config --show channels
channels:
  - defaults
```

Here the full stacktrace

```bash
(base) ➜  benchmark_lasso git:(main) BENCHOPT_DEBUG=1 benchopt install . --env --recreate
Installing 'benchmark_lasso' requirements
Creating conda env 'benchopt_benchmark_lasso':...
conda env config:
----------------------------------------
channels:
  - conda-forge
  - nodefaults
dependencies:
  - python=3.8
  - numpy
  - cython
  - compilers
  - pip
  - pip:
    - -e /home/mathurin/workspace/benchopt
----------------------------------------
------------------------------------------------------------
set -e
conda env create --force -n benchopt_benchmark_lasso -f /tmp/conda_env_l616bqj9.yml
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt --version --check-editable
------------------------------------------------------------
 done
# Install
Collecting packages:
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/blitz.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/cd.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/celer.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/cyanure.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/glmnet.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/glum.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/julia_pgd.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/l_bfgs_b.py Solver
------------------------------------------------------------
- 'L-BFGS-B' already available in 'benchopt_benchmark_lasso'
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/lightning.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/modopt_fista.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/modopt_pogm.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/python_pgd.py Solver
------------------------------------------------------------
- 'Python-PGD' already available in 'benchopt_benchmark_lasso'
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/r_pgd.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/sklearn.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/solvers/spams.py Solver
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/datasets/finance.py Dataset
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/datasets/leukemia.py Dataset
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/datasets/libsvm.py Dataset
------------------------------------------------------------
------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
benchopt check-install /home/mathurin/workspace/benchopt/benchmarks/benchmark_lasso/datasets/simulated.py Dataset
------------------------------------------------------------
- 'Simulated' already available in 'benchopt_benchmark_lasso'
... done
Installing required packages for:
- Blitz
- cd
- Celer
- Cyanure
- glmnet
- glum
- Julia-PGD
- Lightning
- ModOpt-FISTA
- ModOpt-POGM
- R-PGD
- sklearn
- spams
- finance
- leukemia
- libsvm
...------------------------------------------------------------
set -e
# Setup conda
unset R_HOME
eval "$(conda shell.bash hook)"
conda activate benchopt_benchmark_lasso

# Run script
conda install --update-all -y mkl rpy2 scikit-learn numba r-glmnet r-base julia python-spams glum
pip install libsvmdata git+https://github.com/CEA-COSMIC/ModOpt.git julia git+https://github.com/scikit-learn-contrib/lightning.git git+https://github.com/tbjohns/blitzl1.git@master cyanure-mkl celer
------------------------------------------------------------
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - glum
  - python-spams
  - julia

Current channels:

  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.


Traceback (most recent call last):
  File "/home/mathurin/anaconda3/bin/benchopt", line 33, in <module>
    sys.exit(load_entry_point('benchopt', 'console_scripts', 'benchopt')())
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/mathurin/anaconda3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/mathurin/workspace/benchopt/benchopt/cli/main.py", line 274, in install
    benchmark.install_all_requirements(
  File "/home/mathurin/workspace/benchopt/benchopt/benchmark.py", line 233, in install_all_requirements
    install_in_conda_env(
  File "/home/mathurin/workspace/benchopt/benchopt/utils/conda_env_cmd.py", line 181, in install_in_conda_env
    _run_shell_in_conda_env(
  File "/home/mathurin/workspace/benchopt/benchopt/utils/shell_cmd.py", line 130, in _run_shell_in_conda_env
    return _run_shell(
  File "/home/mathurin/workspace/benchopt/benchopt/utils/shell_cmd.py", line 68, in _run_shell
    raise RuntimeError(raise_on_error.format(output=output))
RuntimeError: Failed to conda install packages ('pip:libsvmdata', 'pip:git+https://github.com/CEA-COSMIC/ModOpt.git', 'mkl', 'rpy2', 'pip:julia', 'pip:git+https://github.com/scikit-learn-contrib/lightning.git', 'scikit-learn', 'numba', 'pip:git+https://github.com/tbjohns/blitzl1.git@master', 'r-glmnet', 'r-base', 'julia', 'python-spams', 'pip:cyanure-mkl', 'pip:celer', 'glum')
Error:
(base) ➜  benchmark_lasso git:(main) 
```

### PR DESCRIPTION
**Pull request by mathurinm**
I also set capture_stdout to False, even without CONDA_DEBUG since env creation takes a long time and people want feedback e.g. #90 

Fix #241

### CODE REVIEW

File: benchopt/utils/conda_env_cmd.py [hunk 1]
```diff
@@ -112,8 +112,14 @@ def create_conda_env(env_name, recreate=False, with_pytest=False, empty=False):
     try:
         _run_shell(
             f"{CONDA_CMD} env create {force} -n {env_name} -f {env_yaml.name}",
-            capture_stdout=True, raise_on_error=True
+            capture_stdout=False, raise_on_error=True
```
Comment (tomMoral) [inline benchopt/utils/conda_env_cmd.py:115]: I would actually add an option `-q` to be able to shut this up.
also, you need to update `l.104,136,139` to take this into account

File: benchopt/utils/conda_env_cmd.py [hunk 2]
```diff
@@ -112,8 +112,14 @@ def create_conda_env(env_name, recreate=False, with_pytest=False, empty=False):
...
         )
+        # the channels priorities cannot be set through the yaml file,
+        # we need to do it at the env creation
+        # see https://stackoverflow.com/questions/70098418/
+        _run_shell_in_conda_env(
+            f"{CONDA_CMD} config --env --prepend channels nodefaults " +
+            "--prepend channels conda-forge", env_name, capture_stdout=False)
```
Comment (tomMoral) [inline benchopt/utils/conda_env_cmd.py:122]: I really don't get the conda system but nice that there is an easy solution.
Also, the SO question seems fairly recent so I guess it changed in the latest version? If so, should we start running the CI on multiple version of `conda`?

**Review by tomMoral — COMMENTED** [review]
Nice, a few extra updates for changing the display but it looks good overall

File: benchopt/utils/conda_env_cmd.py (hunk 2 above)
Comment (mathurinm) [reply benchopt/utils/conda_env_cmd.py:122]: no the SO question is me yesterday, I don't know if there is an issue with conda version

Comment (tomMoral) [reply benchopt/utils/conda_env_cmd.py:122]: Aha ok!
then great.

**Comment by mathurinm** [comment]
On my machine `benchopt install . --env --recreate` looks ok with and without `--quiet`

File: benchopt/utils/conda_env_cmd.py [hunk 3]
```diff
@@ -101,19 +105,29 @@ def create_conda_env(env_name, recreate=False, with_pytest=False, empty=False):
     if empty:
         benchopt_env = EMPTY_ENV
 
-    print(f"Creating conda env '{env_name}':...", end='', flush=True)
+    print(f"Creating conda env '{env_name}':... ", end='', flush=True)
```
Comment (tomMoral) [inline benchopt/utils/conda_env_cmd.py:108]:
```suggestion
    if quiet:
        print(f"Creating conda env '{env_name}':... ", end='', flush=True)
```

File: benchopt/utils/conda_env_cmd.py [hunk 4]
```diff
@@ -82,7 +85,8 @@ def create_conda_env(env_name, recreate=False, with_pytest=False, empty=False):
                 "by either using the --recreate option or fixing the version "
                 f"of benchopt in conda env {env_name}."
             )
-        print('done')
+        if quiet:
+            print("done")
```
Comment (tomMoral) [inline benchopt/utils/conda_env_cmd.py:89]: Is this needed?
```suggestion
        print("done")
```

**Review by tomMoral — APPROVED** [review]
LGTM!

**Pull request merged**